File: SHSegmentedControlTableView/Demo/TestCollectionView.m
//
//  TestCollectionView.m
//  SHSegmentedControlTableView
//
//  Created by angle on 2017/10/23.
//  Copyright © 2017年 angle. All rights reserved.
//

#import "TestCollectionView.h"
@interface TestCollectionView ()<UICollectionViewDelegate,UICollectionViewDataSource,UICollectionViewDelegateFlowLayout>

@property (nonatomic, strong) NSArray *colors;


@end

@implementation TestCollectionView

/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/
- (void)setNum:(NSInteger)num {
    _num = num;
    [self reloadData];
}
- (instancetype)initWithFrame:(CGRect)frame collectionViewLayout:(UICollectionViewLayout *)layout {
    if (self = [super initWithFrame:frame collectionViewLayout:layout]) {
        self.backgroundColor = [UIColor whiteColor];
        self.num = 0;
        self.colors = @[[UIColor brownColor],
                        [UIColor grayColor],
                        [UIColor redColor],
                        [UIColor greenColor],
                        [UIColor blueColor],
                        [UIColor cyanColor],
                        [UIColor yellowColor],
                        [UIColor magentaColor],
                        [UIColor orangeColor],
                        [UIColor purpleColor]];
        self.dataSource = self;
        self.delegate = self;
        [self registerClass:UICollectionViewCell.class forCellWithReuseIdentifier:@"cell"];
    }
    return self;
}
#pragma mark -
#pragma mark   ==============UICollectionViewDataSource==============
- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
    return self.num;
}
- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
    UICollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:@"cell" forIndexPath:indexPath];
    cell.backgroundColor = [self.colors objectAtIndex:(arc4random()%self.colors.count)];
    return cell;
}

- (void)collectionView:(UICollectionView *)collectionView didSelectItemAtIndexPath:(NSIndexPath *)indexPath
{
    NSLog(@"%ld-%ld",indexPath.section,indexPath.item);
}
@end

File: Swift/SHSegmentedControlTableView/SHSegmentedControlTableView/TestCollectionView.swift
//
//  TestCollectionView.swift
//  SHSegmentedControlTableView
//
//  Created by angle on 2018/1/30.
//  Copyright © 2018年 angle. All rights reserved.
//

import UIKit

import SHSegmentedControl

class TestCollectionView: SHCollectionView, UICollectionViewDataSource, UICollectionViewDelegate {
    
    open var num:NSInteger!
    
    
    func collectionView(frame: CGRect) -> UICollectionView {
        let layout:UICollectionViewFlowLayout = UICollectionViewFlowLayout.init()
        let collectionView = TestCollectionView.init(frame: frame, collectionViewLayout: layout)
        return collectionView
    }
    
    override init(frame: CGRect, collectionViewLayout layout: UICollectionViewLayout) {
        super.init(frame: frame, collectionViewLayout: layout)
        self.backgroundColor = UIColor.white
        self.num = 0
        self.dataSource = self
        self.delegate = self
        self.register(UITableViewCell.classForCoder(), forCellWithReuseIdentifier: "cell")
        self.allowsSelection = false
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    func getArcDomColor() -> UIColor {
        let colors:NSArray = [UIColor.brown,
                              UIColor.gray,
                              UIColor.red,
                              UIColor.green,
                              UIColor.blue,
                              UIColor.cyan,
                              UIColor.yellow,
                              UIColor.magenta,
                              UIColor.orange,
                              UIColor.purple]
        
        let color:UIColor = colors.object(at: Int(arc4random()%UInt32(colors.count))) as! UIColor
        
        return color
    }

    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        return self.num
    }
    
    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
        let cell:UICollectionViewCell = collectionView.dequeueReusableCell(withReuseIdentifier: "cell", for: indexPath)
        cell.backgroundColor = self.getArcDomColor()
        return cell
    }
}
